spec/models/source_manager_spec.rb

```diff
@@ -0,0 +1,23 @@
+describe SourceManager do
+  subject { SourceManager }
+
+  it { should respond_to(:source) }
+
+  let(:data_structure) do
+    Struct.new(:images) do
+      def images
+        [Struct.new(:src) do
+          def src
+            'http://test-image.jpg'
+          end
+        end.new]
+      end
+    end.new
+  end
+
+  describe '#source' do
+    it 'abstracts source from image url' do
+      expect(SourceManager.source(data_structure)).to eq '//test-image.jpg'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.